sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/FileProperties.java

```diff
@@ -153,7 +152,18 @@ public static FileProperties fromJson(JsonReader jsonReader) throws IOException
                 if ("lastModified".equals(fieldName)) {
                     lastModified = reader.getNullable(nonNullReader -> OffsetDateTime.parse(nonNullReader.getString()));
                 } else if ("contentLength".equals(fieldName)) {
-                    contentLength = reader.getLong();
+                    if (reader.currentToken() == JsonToken.STRING) {
+                        String contentLengthStr = reader.getString();
+                        try {
+                            contentLength = Long.parseLong(contentLengthStr);
+                        } catch (NumberFormatException e) {
+                            throw new IOException("Expected numeric contentLength, but found: " + contentLengthStr, e);
+                        }
+                    } else if (reader.currentToken() == JsonToken.NUMBER) {
+                        contentLength = reader.getLong();
+                    } else {
+                        throw new IOException("Expected contentLength to be a number or string, but found other type");
+                    }
```

Comment:
This seems from a mismatch in TypeSpec source
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/batch/Azure.Batch/models.tsp#L1171-L1187

Here the type is int32

Reply:
We actually found that this was a mistake in our TypeSpec. It is defined as a long (int64) in the service code and the management plane swagger, so we updated the TypeSpec to correctly define it as int64 and not int32: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/batch/Azure.Batch/models.tsp#L1054

Reply:
Got it.

However, the record looks like a JSON String, not a JSON Numeric
https://github.com/Azure/azure-sdk-assets/blob/java/batch/azure-compute-batch_aff1c3044f/java/sdk/batch/azure-compute-batch/src/test/resources/session-records/FileTests.canReadFromTaskFile.json#L259-L283